packages/go/analysis/ad/adcs.go

```diff
@@ -27,13 +27,198 @@ import (
 	"github.com/specterops/bloodhound/dawgs/query"
 	"github.com/specterops/bloodhound/dawgs/util/channels"
 	"github.com/specterops/bloodhound/graphschema/ad"
+	"github.com/specterops/bloodhound/log"
 	"github.com/specterops/bloodhound/slices"
 )
 
 var (
-	ErrNoCertParent = errors.New("cert has no parent")
+	ErrNoCertParent     = errors.New("cert has no parent")
+	EkuAnyPurpose       = "2.5.29.37.0"
+	EkuCertRequestAgent = "1.3.6.1.4.1.311.20.2.1"
 )
 
+func PostEnrollOnBehalfOf(certTemplates []graph.Node, operation analysis.StatTrackedOperation[analysis.CreatePostRelationshipJob]) error {
+	versionOneTemplates := make([]graph.Node, 0)
+	versionTwoTemplates := make([]graph.Node, 0)
+
+	for _, node := range certTemplates {
+		if version, err := node.Properties.Get(ad.SchemaVersion.String()).Int(); err != nil {
+			log.Errorf("Error getting schema version for cert template %d: %v", node.ID, err)
+		} else {
+			if version == 1 {
+				versionOneTemplates = append(versionOneTemplates, node)
+			} else if version >= 2 {
+				versionTwoTemplates = append(versionTwoTemplates, node)
+			}
```

Comment:
It is probably unlikely to reach the `else` case but if we do happen to not meet either of these two conditions, logging some information would be good for debugging.